test/automated-suite.js

```diff
@@ -0,0 +1,217 @@
+/*
+  Copyright 2016 Google Inc. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+'use strict';
+
+// This is a test and we want descriptions to be useful, if this
+// breaks the max-length, it's ok.
+
+var fs = require('fs');
+var webdriver = require('selenium-webdriver');
+var chrome = require('selenium-webdriver/chrome');
+var firefox = require('selenium-webdriver/firefox');
+require('chai').should();
+
+/* eslint-disable max-len, no-console, padded-blocks, no-multiple-empty-lines */
+/* eslint-env node,mocha */
+
+// These tests make use of selenium-webdriver. You can find the relevant
+// documentation here: http://selenium.googlecode.com/git/docs/api/javascript/index.html
+
+describe('Test SW-Toolbox', () => {
+  // Driver is initialised to null to handle scenarios
+  // where the desired browser isn't installed / fails to load
+  // Null allows afterEach a safe way to skip quiting the driver
+  let driver = null;
+
+  afterEach(done => {
+    if (driver === null) {
+      return done();
+    }
+
+    driver.quit()
+    .then(() => {
+      driver = null;
+      done();
+    })
+    .thenCatch(() => {
+      driver = null;
+      done();
+    });
+  });
+
+  let checkFileExists = path => {
+    return new Promise((resolve, reject) => {
+      fs.stat(path, err => {
+        if (err) {
+          return reject(err);
+        }
+        resolve();
+      });
+    });
+  };
+
+  let performTests = (browserName, driver) => {
+    // The driver methods are wrapped in a new promise because the
+    // selenium-webdriver API seems to using some custom promise
+    // implementation that has slight behaviour differences.
+    return new Promise((resolve, reject) => {
+      driver.get('http://localhost:8888/test/')
+      .then(() => {
+        return driver.executeScript('return window.navigator.userAgent;');
+      })
+      .then(userAgent => {
+        // This is just to help with debugging so we can get the browser version
+        console.log('    Browser User Agent [' + browserName + ']: ' + userAgent);
+      })
+      .then(() => {
+        // We get webdriver to wait until window.swtoolbox.testResults is defined.
+        // This is set in the in browser mocha tests when the tests have finished
+        // successfully
+        return driver.wait(function() {
```

Comment:
Putting in an explicit timeout set to whatever you think is reasonable would be clearer than relying on whatever the default timeout value is.